### PR TITLE
Repo setup for Azure

### PR DESCRIPTION
**Pull request by raghavanandan**
This PR is for setting up the repo for Azure terraform. This includes changes/inclusion to,

1. Terraform module structure
2. Github actions for PR release
3. A place-holder service-principal module
4. README.md

`terraform plan` output:

```
var.subscription_id
  Subscription ID in which to create a trust relationship

  Enter a value: 91987989-c0bc-4558-9151-cd308aa333e6

data.azurerm_subscription.primary: Reading...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform planned the following actions, but then encountered a problem:

  # azuread_service_principal.sysdig_sp will be created
  + resource "azuread_service_principal" "sysdig_sp" {
      + account_enabled             = true
      + app_role_ids                = (known after apply)
      + app_roles                   = (known after apply)
      + application_id              = "0a800625-0e0d-4805-827c-743371e8518c"
      + application_tenant_id       = (known after apply)
      + display_name                = (known after apply)
      + homepage_url                = (known after apply)
      + id                          = (known after apply)
      + logout_url                  = (known after apply)
      + oauth2_permission_scope_ids = (known after apply)
      + oauth2_permission_scopes    = (known after apply)
      + object_id                   = (known after apply)
      + redirect_uris               = (known after apply)
      + saml_metadata_url           = (known after apply)
      + service_principal_names     = (known after apply)
      + sign_in_audience            = (known after apply)
      + tags                        = (known after apply)
      + type                        = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

### CODE REVIEW

File: modules/services/service-principal/main.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
+provider "azurerm" {
```
Comment (mdkulkarni15) [inline modules/services/service-principal/main.tf:1]: This might need some work later, once we start working on it.

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+* @draios/team-secure-onboarding
```
Comment (mdkulkarni15) [inline CODEOWNERS:1]: this one has some errors

Comment (mdkulkarni15) [reply CODEOWNERS:1]: does our team have write access to the repo?

Comment (mdkulkarni15) [reply CODEOWNERS:1]: Unknown owner on line 1: make sure the team @draios/team-secure-onboarding exists, is publicly visible, and has write access to the repository
* @draios/team-secure-onboarding

Comment (mdkulkarni15) [reply CODEOWNERS:1]: same error is on the gcp repo btw

Comment (raghavanandan) [reply CODEOWNERS]: Fixed!

**Review by mdkulkarni15 — APPROVED** [review]

**Review by cgeers — APPROVED** [review]

**Pull request merged**